src/types/errors.ts

```diff
@@ -25,3 +25,19 @@ export interface ErrorPayload {
   code: AppErrors
   message: string
 }
+
+// Adds strict type-check that a type was exhausted
+// https://www.typescriptlang.org/docs/handbook/2/narrowing.html#exhaustiveness-checking
+// https://stackoverflow.com/questions/41102060/typescript-extending-error-class
+export class ExhaustedTypeError extends Error {
```

Comment:
This is the same as https://github.com/oasisprotocol/oasis-wallet-web/blob/71c3967b865812d84faad783395c09c9aebfe3f6/src/types/errors.ts#L38